include/libvfio-user.h

```diff
@@ -620,6 +664,10 @@ vfu_ctx_try_attach(vfu_ctx_t *vfu_ctx);
 /*
  * FIXME need to make sure that there can be at most one capability with a given
  * ID, otherwise this function will return the first one with this ID.
+ *
+ * FIXME do we want this? It's only used right now to get MSI-X enabled by
```

Comment:
There can be multipe capabilities with the same ID, the vendor-specific capability is one of them, not sure about others.

Not sure what's best TBH. E.g. in the SPDK case it's used only to access `msix->mxc.mxe`. We can introduce function that access particular bits of capabilities, e.g. `vfu_pci_cap_msix_is_enabled(vfu_ctx_t *vfu_ctx)` for the SPDK case, however each capabilities has many individual fields and there are a couple of dozens capabilities, so we might easily end up with a hundend functions just for capabilities. Since we don't yet know enough we might allow direct access to the capabilities for now and if it turns out that an API is better then we can deprecate `vfu_ctx_get_cap`.

Reply:
Yeah, I think you're probably right. But we could potentially make this be a helper against the config space pointer, depends what we do above.